snprc_ehr/src/client/SndEventsWidget/SndEventsWidget.tsx

```diff
@@ -62,24 +63,24 @@ export const SndEventsWidget: FC<Props> = memo((props: Props) => {
 
     return (
         <div>
-            {!hasPermission && (
+            {!hasReadPermission && (
                 <Alert>User Does not have permission to view this panel</Alert>
             )}
-            {hasPermission && (filterConfig === undefined || filterConfig.length == 0) && (
+            {hasReadPermission && (filterConfig === undefined || filterConfig.length == 0) && (
                 form()
             )
             }
-            {hasPermission && (filterConfig !== undefined && filterConfig.length != 0 && filterConfig?.filters.inputType === 'none') && (
+            {hasReadPermission && (filterConfig !== undefined && filterConfig.length != 0 && filterConfig?.filters.inputType === 'none') && (
                 <Alert>'Entire Database' filter is not supported for this query.</Alert>
             )}
-            {hasPermission && subjectIds[0] === 'none' && (
+            {hasReadPermission && subjectIds[0] === 'none' && (
                 <Alert>No animals were found for filter selections</Alert>
             )}
             {message && (
                 <Alert bsStyle={status}>{message}</Alert>
             )}
-            {hasPermission && subjectIds && (
-                <EventListingGridPanel subjectIDs={subjectIds} onChange={handleUpdateResponse} onError={handleError}/>
+            {hasReadPermission && subjectIds && (
+                <EventListingGridPanel subjectIDs={subjectIds} onChange={handleUpdateResponse} onError={handleError} hasWritePermission={hasWritePermission}/>
```

Comment:
How are we using this hasWritePermission here Jay? Is this for the Clinical -> SND Events window Tab?

Reply:
Yes hasWritePermission is used to conditionally display the pencil and trash icons on the widget SNDEventsWidget grid ID column. Will not display those icons if hasWritePermission is false